pkg/rosa/client_test.go

```diff
@@ -0,0 +1,139 @@
+package rosa
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	rosacontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/rosa/api/v1beta2"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/scope"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/logger"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/util/system"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/onsi/gomega"
+)
+
+func createROSAControlPlaneScope(wlSecret, mgrSecret *corev1.Secret, cp *rosacontrolplanev1.ROSAControlPlane) *scope.ROSAControlPlaneScope {
```

Comment:
better to create another function named `createFakeClientWithObjects` and give it the obj (secrets) as param return a fakeClient  then use it to `createROSAControlPlaneScope`

Reply:
Nah, that would be just one more function created just for tests. Maybe in the future, if we'll need to call it separately, then we can extract it, but I'd rather leave it as it is right now.

I simplified the `createROSAControlPlaneScope()` function and made it a varadic fn. That way the function is much shorter.